.travis.yml

```diff
@@ -21,91 +21,121 @@ install:
         - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12
 
-    - name: spark-3.0_micro
-      before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+    - name: hibench_spark-3.0_hadoop-2.7_micro
       script:
-          - cp ./travis/benchmarks_micro.lst ./conf/benchmarks.lst
-          - ./travis/build_and_run_all_spark_3.0.sh
+        - ls -l ./travis
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12
+        - cp ./travis/benchmarks_micro.lst ./conf/benchmarks.lst
+        - source ./travis/set_spark-3.0_hadoop-2.7_env.sh
 
-    - name: spark-3.0_sql
-      before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+    - name: hibench_spark-3.0_hadoop-2.7_sql
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_sql.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/set_spark-3.0_hadoop-2.7_env.sh
+
 
-    - name: spark-3.0_websearch
-      before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+    - name: hibench-spark-3.0_hadoop-2.7_websearch
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_websearch.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/set_spark-3.0_hadoop-2.7_env.sh
 
-    - name: spark-3.0_ml
-      before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+    - name: hibench_spark-3.0_hadoop-2.7_ml
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_ml.lst ./conf//benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/set_spark-3.0_hadoop-2.7_env.sh
 
-    - name: spark-3.0_graph
-      before_script:
-        - ./travis/export_environment_variables_spark_3.0.sh
+    - name: hibench_spark-3.0_hadoop-2.7_graph
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=3.0 -Dscala=2.12 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_graph.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_3.0.sh
+        - source ./travis/set_spark-3.0_hadoop-2.7_env.sh
 
-    - name: spark-2.4_micro
-      before_script:
-        - ./travis/export_environment_variables_spark_2.4.sh
+    - name: hibench_spark-2.4_hadoop-2.7_micro
       script:
+        - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.4 -Dscala=2.11 -Dhive=0.14 -Dhadoop=2.4
         - cp ./travis/benchmarks_micro.lst ./conf/benchmarks.lst
-        - ./travis/build_and_run_all_spark_2.4.sh
+        - source ./travis/set_spark-2.4_hadoop-2.7_env.sh
```

Comment:
Let's keep the original structure, the flow is very clear:
- export environment variables
- copy benchmark.list
- build & run